### PR TITLE
adding compile time assertion support whenever possible

### PR DESCRIPTION
**Pull request by devnexen**
and convert some runtime which deserve to be evaluated earlier.

### CODE REVIEW

File: src/common/header/shared.h [hunk 1]
```diff
@@ -77,6 +78,8 @@ typedef unsigned char byte;
 	#define YQ2_ATTR_NORETURN       __attribute__ ((noreturn))
 	#define YQ2_ATTR_MALLOC         __attribute__ ((__malloc__))
 	#define YQ2_ATTR_INLINE         __attribute__((always_inline)) inline
+    	// GCC supports this extension since 4.6
```
Comment (DanielGibson) [inline src/common/header/shared.h:81]: indenting is inconsistent with surrounding lines

**Comment by DanielGibson** [comment]
good idea generally, though I'd call it `YQ2_STATIC_ASSERT`

**Comment by devnexen** [comment]
I know but I tried to avoid long names :-) but I ll change it no pb.

**Comment by Yamagi** [comment]
Thank you :)

**Pull request merged**